addons/product/models/product_category.py

```diff
@@ -63,3 +63,6 @@ def _unlink_except_default_category(self):
         expense_category = self.env.ref('product.cat_expense', raise_if_not_found=False)
         if expense_category and expense_category in self:
             raise UserError(_("You cannot delete the %s product category.", expense_category.name))
+        saleable_category = self.env.ref('product.product_category_1', raise_if_not_found=False)
+        if saleable_category and saleable_category in self:
+            raise UserError(_("You cannot delete this %s product category.", saleable_category.name))
```

Comment:
(You forgot one xmlid in your list)
It would be cleaner, indeed. But you would then lose the blame of the existing lines and people don't always like that...
Honestly it's not that important, I let you see that with the final reviewer

Reply:
This method is not in master anymore, so I don't think it deserves a refactoring.